comp/otelcol/converter/impl/coreconfig.go

```diff
@@ -0,0 +1,150 @@
+			if !ok {
+				ddprofilingMap = make(map[string]any)
+				extensionMap[extension] = ddprofilingMap
+			}
+			profilerOptions, ok := ddprofilingMap["profiler_options"]
+			if !ok || profilerOptions == nil {
+				profilerOptions = make(map[string]any)
+				ddprofilingMap["profiler_options"] = profilerOptions
+			}
+			profilerOptionsMap, ok := profilerOptions.(map[string]any)
+			if !ok {
+				return
+			}
+			if profilerOptionsMap["env"] != nil && profilerOptionsMap["env"] != "" {
+				return
+			}
+			if coreCfg.Get("env") == nil || coreCfg.Get("env") == "" {
+				return
+			}
+			profilerOptionsMap["env"] = coreCfg.Get("env")
+		}
+	}
+	*conf = *confmap.NewFromStringMap(stringMapConf)
+}
+
+// addAPIKeySite adds the API key and site from core config to github.com/open-telemetry/opentelemetry-collector-contrib/pkg/datadog/config.APIConfig.
+func addAPIKeySite(conf *confmap.Conf, coreCfg config.Component, compType string, compName string) {
+	stringMapConf := conf.ToStringMap()
+	components, ok := stringMapConf[compType]
+	if !ok {
+		return
+	}
+	componentMap, ok := components.(map[string]any)
+	if !ok {
+		return
+	}
+	reg, err := regexp.Compile(secretRegex)
+	if err != nil {
+		return
+	}
+	for component := range componentMap {
+		if componentName(component) == compName {
+			datadog, ok := componentMap[component]
+			if !ok {
+				return
+			}
+			datadogMap, ok := datadog.(map[string]any)
+			if !ok {
+				// datadog section is there, but there is nothing in it. We
+				// need to add it so we can add to it.
+				componentMap[component] = make(map[string]any)
+				datadogMap = componentMap[component].(map[string]any)
+			}
+			api, ok := datadogMap["api"]
+			if !ok && compName == "ddprofiling" {
+				// in otel-agent path, ddprofiling does not need api key/site,
+				// so it's expected not to be present. don't add it.
+				return
+			}
+			// ok can be true if api section is there but contains nothing (api == nil).
+			// In which case, we need to add it so we can add to it.
+			if !ok || api == nil {
+				datadogMap["api"] = make(map[string]any, 2)
+				api = datadogMap["api"]
+			}
+			apiMap, ok := api.(map[string]any)
+			if !ok {
+				return
+			}
+
+			// api::site
+			apiSite := apiMap["site"]
+			if (apiSite == nil || apiSite == "") && coreCfg.Get("site") != nil {
+				apiMap["site"] = coreCfg.Get("site")
+			} else if (apiSite == nil || apiSite == "") && coreCfg.Get("site") == nil {
+				// if site is nil or empty string, and core config site is unset, set default
+				// site. Site defaults to an empty string in helm chart:
+				// https://github.com/DataDog/helm-charts/blob/datadog-3.86.0/charts/datadog/templates/_otel_agent_config.yaml#L24.
+				apiMap["site"] = "datadoghq.com"
+			}
+
+			// api::key
+			var match bool
+			apiKey, ok := apiMap["key"]
+			if ok {
+				var key string
+				if keyString, okString := apiKey.(string); okString {
+					key = keyString
+				}
+				if ok && key != "" {
```

Comment:
```suggestion
				if key != "" {
```
`ok` can be removed here since it's already in the `if ok`